primitives/session-manager/Cargo.toml

```diff
@@ -11,9 +11,11 @@ frame-system = { workspace = true }
 log = { workspace = true }
 pallet-session = { workspace = true }
 pallet-session-validator-management = { workspace = true }
+sp-session-validator-management = { workspace = true, optional = true }
```

Comment:
This is not a new dependency. It was a transitive dependency before. Without this line it was impossible to build this (session-manager) crate without building the whole project: `cargo build -p session-manager` was failing on std/no-std configuration errors. All changes in this file are related to this issue.

Reply:
This happens very often, maybe we should have a step checking if all the crates build solo, but this would probably cost too much time. :\